test/mjsunit/asm/int32x4.js

```diff
@@ -0,0 +1,418 @@
+  var i4and = i4.and;
+  var i4sub = i4.sub;
+  var i4mul = i4.mul;
+  var i4or = i4.or;
+  var i4xor = i4.xor;
+  
+  var a = i4(imports.a);
+  var b = i4(imports.b);
+
+  function add(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4add(a, b);
+    return i4(ret);
+  }
+
+  function addLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4add(a, b);
+    return i4(ret);
+  }
+
+  function addImports() {
+    var ret = i4();
+    ret = i4add(a, b);
+    return i4(ret);
+  }
+
+  function sub(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4sub(a, b);
+    return i4(ret);
+  }
+
+  function subLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4sub(a, b);
+    return i4(ret);
+  }
+
+  function subImports() {
+    var ret = i4();
+    ret = i4sub(a, b);
+    return i4(ret);
+  }
+
+  function mul(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4mul(a, b);
+    return i4(ret);
+  }
+
+  function mulLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4mul(a, b);
+    return i4(ret);
+  }
+
+  function mulImports() {
+    var ret = i4();
+    ret = i4mul(a, b);
+    return i4(ret);
+  }
+
+  function and(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4and(a, b);
+    return i4(ret);
+  }
+
+  function andLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4and(a, b);
+    return i4(ret);
+  }
+
+  function andImports() {
+    var ret = i4();
+    ret = i4and(a, b);
+    return i4(ret);
+  }
+
+  function or(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4or(a, b);
+    return i4(ret);
+  }
+
+  function orLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4or(a, b);
+    return i4(ret);
+  }
+
+  function orImports() {
+    var ret = i4();
+    ret = i4or(a, b);
+    return i4(ret);
+  }
+
+  function xor(a, b) {
+    a = i4(a);
+    b = i4(b);
+    var ret = i4();
+    ret = i4xor(a, b);
+    return i4(ret);
+  }
+
+  function xorLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var b = i4(+5, +6, +7, +8);
+    var ret = i4();
+    ret = i4xor(a, b);
+    return i4(ret);
+  }
+
+  function xorImports() {
+    var ret = i4();
+    ret = i4xor(a, b);
+    return i4(ret);
+  }
+
+  function getx(a) {
+    a = i4(a);
+    var x = a.x;
+    return x | 0;
+  }
+
+  function gety(a) {
+    a = i4(a);
+    var y = a.y;
+    return y | 0;
+  }
+
+  function getz(a) {
+    a = i4(a);
+    var z = a.z;
+    return z | 0;
+  }
+
+  function getw(a) {
+    a = i4(a);
+    var w = a.w;
+    return w | 0;
+  }
+
+  function getxLocal() {
+    var a = i4(+1, +2, +3, +4);
+    var x = a.x;
+    return x | 0;
+  }
+
+  function getyLocal(a) {
```

Comment:
no parameter "a", use local variable "a". 

Seems the same issue in float64x2.js and I will fix it.